test/suites/azuregarbagecollection/suite_test.go

```diff
@@ -0,0 +1,76 @@
+/*
+Portions Copyright (c) Microsoft Corporation.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package azuregarbagecollection
+
+import (
+	"testing"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/samber/lo"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/network/armnetwork"
+	"github.com/Azure/karpenter-provider-azure/pkg/apis/v1alpha2"
+	azkarptest "github.com/Azure/karpenter-provider-azure/pkg/test"
+	"github.com/Azure/karpenter-provider-azure/test/pkg/environment/azure"
+	karpv1 "sigs.k8s.io/karpenter/pkg/apis/v1"
+)
+
+var env *azure.Environment
+var nodeClass *v1alpha2.AKSNodeClass
+var nodePool *karpv1.NodePool
+
+func TestGC(t *testing.T) {
+	RegisterFailHandler(Fail)
+	BeforeSuite(func() {
+		env = azure.NewEnvironment(t)
+	})
+	RunSpecs(t, "Azure Garbage Collection")
+}
+
+var _ = BeforeEach(func() {
+	env.BeforeEach()
+	nodeClass = env.DefaultAKSNodeClass()
+	nodePool = env.DefaultNodePool(nodeClass)
+})
+var _ = AfterEach(func() { env.Cleanup() })
+var _ = AfterEach(func() { env.AfterEach() })
+
+var _ = Describe("gc", func() {
+	It("should garbage collect network interfaces created by karpenter", func() {
+		env.ExpectCreatedInterface(armnetwork.Interface{
+			Name: lo.ToPtr("orphan-nic"),
+			Location: lo.ToPtr(env.Region),
+			Tags: azkarptest.ManagedTags("default"),
+			Properties: &armnetwork.InterfacePropertiesFormat{
+				IPConfigurations: []*armnetwork.InterfaceIPConfiguration{
+					{
+						Name: lo.ToPtr("ip-config"), 
+						Properties: &armnetwork.InterfaceIPConfigurationPropertiesFormat{
+							Primary: lo.ToPtr(true),
+							Subnet: env.GetClusterSubnet(), 
+							PrivateIPAllocationMethod: lo.ToPtr(armnetwork.IPAllocationMethodDynamic),
+						},
+					},			
+				},
+
+			},
+		})	
```

Comment:
I will refactor this to use **https://github.com/Azure/karpenter-provider-azure/blob/main/pkg/test/networkinterfaces.go#L39** in a followup pr. 

First to use those options we need to remove some assumptions around the codebase that call our subscriptionIDs "subscriptionID" and there is other general cleanup to be done there(removing nodepool smell from InterfaceOptions). 

This is already compact enough for our purposes.